### PR TITLE
ci: add DAST scan workflow

### PR DESCRIPTION
**Pull request by pmoscode**
## Description

This PR introduce a DAST API scan with ZAP (https://github.com/zaproxy/zaproxy)

Currently there is no authentication done. This should simulate a "normal" attack against the API.

The app is deployed in a local KinD cluster and the ZAP scan is running outside it - as a standalone docker container.
The MIW app is exposed via `kubectl port-forward`, so that ZAP can access it.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by pmoscode** [comment]
Hi @borisrizov-zf, 
can you review please?

File: .github/workflows/dast-scan.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          echo "IP_ADDR=$IP_ADDR" >> $GITHUB_ENV
+
+          POD_NAME=$(kubectl get pods --namespace apps -l "app.kubernetes.io/name=managed-identity-wallet,app.kubernetes.io/instance=testing" -o jsonpath="{.items[0].metadata.name}")
+          CONTAINER_PORT=$(kubectl get pod --namespace apps $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+
+          echo "Port-forwarding 0.0.0.0:8080 to $POD_NAME:$CONTAINER_PORT..."
+
+          kubectl --namespace apps port-forward $POD_NAME 8080:$CONTAINER_PORT --address 0.0.0.0 &
+
+      - name: Sleep for 4s
```
Comment (borisrizov-zf) [inline .github/workflows/dast-scan.yaml:96]: You might add a comment explaining why we wait for exactly 4s

**Review by borisrizov-zf — APPROVED** [review]

File: .github/workflows/dast-scan.yaml (hunk 1 above)
Comment (pmoscode) [reply .github/workflows/dast-scan.yaml:97]: Done.

File: .github/workflows/dast-scan.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
+# /********************************************************************************
+# * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
Comment (carslen) [inline .github/workflows/dast-scan.yaml:2]: As this is a new file, copyright cannot start in the past.

**Review by carslen — APPROVED** [review]

File: .github/workflows/dast-scan.yaml (hunk 2 above)
Comment (pmoscode) [reply .github/workflows/dast-scan.yaml]: Header updated

**Pull request merged**